sklearn/linear_model/cd_fast.pyx

```diff
@@ -143,6 +152,100 @@ cdef extern from "cblas.h":
     void sscal "cblas_sscal"(int N, float alpha, float *X, int incX) nogil
 
 
+# Function to compute the duality gap
+cdef inline floating enet_duality_gap( # Data
```

Comment:
I am fond of the refactoring induced by the new enet_duality_gap functions by the way, it makes the functions more readable. It would be great to add a reference that derives the duality gap that is used for future reference. In another PR though.